[Widgets]: Add missing parameterless constructors and fix docs API defaults

defymecobra opened the pull request:
## Problem

After merging PR #1919, several widgets were missing parameterless constructors required for:
- Reflection-based default value comparison during serialization
- Documentation API default value generation (`TypeUtils.TryToActivate`)

Additionally, the docs API section was not displaying default values because `TryToActivate` only searched for public constructors.

## Solution

### 1. Added internal parameterless constructors to 15 widgets:

Animation, Empty, Fragment, Html, Icon, Json, Kbd, Markdown, Xml, Table, Embed, Loading, Skeleton, Spacer, Terminal

These constructors are `internal` to:
- Enable reflection-based default value extraction
- Prevent external misuse (users should use primary constructors or extension methods)

### 2. Fixed `TryToActivate` in `TypeUtils.cs`:

Added support for non-public constructors using `BindingFlags.NonPublic`:
```csharp
var nonPublicConstructor = type.GetConstructor(
    BindingFlags.Instance | BindingFlags.NonPublic,
    null, Type.EmptyTypes, null);
```

### 3. Added `InternalsVisibleTo` in `AssemblyInfo.cs`:

Exposed internal constructors to docs and samples assemblies:
```csharp
[assembly: InternalsVisibleTo("Ivy.Docs.Shared")]
[assembly: InternalsVisibleTo("Ivy.Samples.Shared")]
```

## Verification

- ✅ Build passes without errors or warnings
- ✅ Docs API section displays default values correctly
- ✅ No breaking changes

rorychatt commented on Ivy/Widgets/Primitives/Loading.cs:
```diff
@@ -5,6 +5,7 @@ namespace Ivy;
 
 public record Loading : WidgetBase<Loading>
 {
+    public Loading() { }
```
why somewhere it is `internal`, somewhere `public`?

defymecobra replied:
Outdated commentary, already fixed this

defymecobra commented:
- Changed public parameterless constructors to internal for Terminal, Loading, Skeleton, Spacer
- Added InternalsVisibleTo for Ivy.Docs.Shared and Ivy.Samples.Shared to allow docs and samples to use internal constructors

rorychatt commented on Ivy/Widgets/Primitives/AppHost.cs:
```diff
@@ -12,7 +12,9 @@ public AppHost(string appId, string? appArgs, string? parentId)
     }
 
-    [Prop] public string AppId { get; set; }
+    internal AppHost() { }
```
Not a widget, don't think we need ctor here

defymecobra commented:
- Removed useless ctor in AppHost
- Updated description for pull-request
